spec/lucky/base_http_client_spec.cr

```diff
@@ -10,58 +10,120 @@ Spec.before_each do
+        with_fake_server(path: "/hello", response_body: "world") do
+          response = MyClient.new.exec(HelloWorldAction, foo: "bar")
 
-        response.body.should eq "world"
+          request = server.last_request
+          request.body.not_nil!.gets_to_end.should eq({foo: "bar"}.to_json)
+        end
       end
     end
-  end
 
-  {% for method in [:put, :patch, :post] %}
+    describe "with a Lucky::RouteHelper" do
+      it "uses the method and path" do
+        with_fake_server(path: "/hello", response_body: "world") do
+          response = MyClient.new.exec(HelloWorldAction.route)
+
+          request = server.last_request
+          request.path.should eq "/hello"
+          request.method.should eq("POST")
+          request.body.not_nil!.gets_to_end.should eq(NamedTuple.new.to_json)
+          response.body.should eq "world"
+        end
+      end
+
+      it "allows passing params" do
+        with_fake_server(path: "/hello", response_body: "world") do
+          response = MyClient.new.exec(HelloWorldAction.route, foo: "bar")
+
+          request = server.last_request
+          request.body.not_nil!.gets_to_end.should eq({foo: "bar"}.to_json)
+        end
+      end
+    end
+  end
 
+  {% for method in [:put, :patch, :post, :delete, :get] %}
     describe "\#{{method.id}}" do
       it "sends correct request to correct uri and gives the correct response" do
-        TestServer.route("hello", "world")
-        Lucky::Server.temp_config(host: "localhost", port: 6226) do
-          client = Lucky::BaseHTTPClient.new
-          response = client.{{method.id}}(
+        with_fake_server(path: "hello", response_body: "world") do
+          response = MyClient.new.{{method.id}}(
             path: "hello",
-            body: { "foo" => "bar" }
+            foo: "bar"
           )
+
           response.body.should eq "world"
+          request = server.last_request
+          request.method.should eq({{ method.id.stringify }}.upcase)
+          request.path.should eq "hello"
+          request.body.not_nil!.gets_to_end.should eq({foo: "bar"}.to_json)
+        end
+      end
 
-          request = server.last_request.not_nil!
-          body = request.body.not_nil!
-          body.gets_to_end.should eq HTTP::Params.encode({ "foo" => "bar" })
+      it "works without params" do
+        with_fake_server(path: "hello", response_body: "world") do
+          response = MyClient.new.{{method.id}}(path: "hello")
+
+          response.body.should eq "world"
+          request = server.last_request
+          request.method.should eq({{ method.id.stringify }}.upcase)
+          request.path.should eq "hello"
+          request.body.not_nil!.gets_to_end.should eq(NamedTuple.new.to_json)
         end
       end
     end
-
   {% end %}
 end
 
+private def with_fake_server(path : String, response_body : String)
+  TestServer.route(path: path, response_body: response_body)
+  Lucky::Server.temp_config(host: "localhost", port: 6226) do
```

Comment:
Any reason behind the `6226` port? Could we name it in a constant? Does it need to be configurable?

Reply:
Extracted a private method. It's the port  of the test server